source/common/http/http2/codec_impl.cc

```diff
@@ -1562,9 +1573,19 @@ ClientConnectionImpl::ClientConnectionImpl(
                                           client_http2_options.options());
   http2_session_factory.init(session_, base(), http2_options);
   allow_metadata_ = http2_options.allow_metadata();
+  idle_session_requires_ping_interval_ = std::chrono::milliseconds(PROTOBUF_GET_MS_OR_DEFAULT(
+      http2_options.connection_keepalive(), connection_idle_interval, 0));
 }
 
 RequestEncoder& ClientConnectionImpl::newStream(ResponseDecoder& decoder) {
+  // If the connection has been idle long enough to trigger a ping, send one
+  // ahead of creating the stream.
+  if (idle_session_requires_ping_interval_.count() != 0 &&
```

Comment:
nit: ditto about zero() (but maybe this is a windmill tilt and count() == 0 is just idiomatic)